src/journal-remote/journal-remote-main.c

```diff
@@ -210,10 +238,18 @@ static int process_http_upload(
                   __func__, connection, *upload_data_size);
 
         if (*upload_data_size) {
+                size_t buf_size;
+                _cleanup_free_ char *buf = NULL;
```

Comment:
Please move them in the `if` branch, and call `journal_importer_push_data()` with the buffer in the branch.
```c
if (source->compression != COMPRESSION_NONE) {
        _cleanup_free_ char *buf = NULL;
        size_t buf_size;

        r = decompress_blob(...);
        if (r < 0)
                return mhd_respond_oom(connection);

        r = journal_importer_push_data(&source->importer, buf, buf_size);
} else
        r = journal_importer_push_data(&source->importer, upload_data, *upload_data_size);
if (r < 0)
        return mhd_respond_oom(connection);
```